### PR TITLE
Man: Use ArgsParser to parse arguments

### LINKED ISSUE #1236: Man: a minor problem in the usage string
@bugaevc @awesomekling I think it is better to change
https://github.com/SerenityOS/serenity/blob/0bb4111735a2fc22487f11f11ba27c00a1fc2366/Userland/man.cpp#L52
to
```C++
fprintf(stderr, "Usage:\t%s <name>\n\t%s [section] <name>\n", argv[0], argv[0]);
```

### PR DESCRIPTION
**Pull request by howar6hill**
@bugaevc Use ArgsParser to parse arguments in the man program. Fix #1236.

### CODE REVIEW

File: Userland/man.cpp [hunk 1]
```diff
@@ -25,6 +25,7 @@
  */
 
 #include <AK/ByteBuffer.h>
+#include <LibCore/ArgsParser.h>
 #include <AK/String.h>
 #include <LibCore/File.h>
 #include <LibMarkdown/MDDocument.h>
```
Comment (bugaevc) [inline Userland/man.cpp]: Headers should be sorted (use `clang-format`)

File: Userland/man.cpp [hunk 2]
```diff
@@ -45,43 +46,39 @@ int main(int argc, char* argv[])
...
+    const char* name_str = nullptr;
+    int section = -1;
 
-    if (argc < 2 || argc > 3) {
-        fprintf(stderr, "Usage:\t%s <name>\n\t%s <section <name>\n", argv[0], argv[0]);
-        exit(1);
-    }
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(section, "section of the man page", "section", Core::ArgsParser::Required::No);
+    args_parser.add_positional_argument(name_str, "name of the man page", "name");
```
Comment (bugaevc) [inline Userland/man.cpp]: Help strings should start with a capital letter, e.g. `"Name of the man page"`

File: Userland/man.cpp [hunk 3]
```diff
@@ -45,43 +46,39 @@ int main(int argc, char* argv[])
...
 
-    if (argc == 2) {
-        name = argv[1];
-    } else {
-        section = argv[1];
-        name = argv[2];
-    }
+    args_parser.parse(argc, argv);
+
+    String name = name_str;
```
Comment (bugaevc) [inline Userland/man.cpp]: Do we need it as a `String` below for something?

The whole point of `Core::ArgsParser` using `const char*`s and not `String`s was that you can just reference your argv and avoid copying the string out to the heap. Sometimes you really want it as a `String`, so you convert it like you did; but most of the times you actually don't need that.

File: Userland/man.cpp [hunk 4]
```diff
@@ -45,43 +46,39 @@ int main(int argc, char* argv[])
 
     unveil(nullptr, nullptr);
 
-    String name;
-    String section;
+    const char* name_str = nullptr;
+    int section = -1;
```
Comment (bugaevc) [inline Userland/man.cpp]: No, man sections are actually strings, it just so happens that most of them (currently on Serenity, all of them) look like ints. But there could be sections like `1p` or whatever.

File: Userland/man.cpp [hunk 5]
```diff
@@ -45,43 +46,39 @@ int main(int argc, char* argv[])
...
+    if (section == -1) {
+        int sections[] = {
+            1,
+            2,
+            3,
+            4,
+            5,
+            6,
+            7,
+            8
```
Comment (bugaevc) [inline Userland/man.cpp]: Ditto

File: Userland/man.cpp [hunk 6]
```diff
@@ -102,9 +99,9 @@ int main(int argc, char* argv[])
 
     dbg() << "Loading man page from " << file->filename();
     auto buffer = file->read_all();
-    String source { (const char*)buffer.data(), (size_t)buffer.size() };
+    String source{ (const char*)buffer.data(), (size_t)buffer.size() };
```
Comment (bugaevc) [inline Userland/man.cpp]: :cry:

Comment (howar6hill) [reply Userland/man.cpp]: @bugaevc Seems .clang-format said it should be so.

Comment (howar6hill) [reply Userland/man.cpp]: @bugaevc tried multiple times.

Comment (awesomekling) [reply Userland/man.cpp]: Please use `clang-format` version 8 or newer.

Comment (howar6hill) [reply Userland/man.cpp]: looks like my clang-format is too old. Upgrade clang-format. It is OK now.

**Comment by howar6hill** [comment]
@bugaevc Fixed the mentioned issues.

**Comment by bugaevc** [comment]
:+1:, now squash the commits please

**Comment by howar6hill** [comment]
@bugaevc commits squashed

**Pull request merged**